rocketchat-core/src/main/java/com/rocketchat/core/callback/MessageCallback.java

```diff
@@ -3,19 +3,22 @@
 import com.rocketchat.common.listener.Callback;
 import com.rocketchat.common.listener.Listener;
 import com.rocketchat.core.model.Message;
-
-import java.lang.reflect.Type;
+import java.util.List;
 
 /**
  * Created by sachin on 22/7/17.
  */
 
 public class MessageCallback {
-    public interface SubscriptionCallback extends Listener {
+    public interface SubscriptionListener extends Listener {
```

Comment:
This is a callback, it is a one shot callback for subscriptions, it is removed from the StreamMiddleware as soon as it is called.

Reply:
This is not a callback and is not removed when called from stream-middleware. I think there is some confusion with other type of other callback :P
This is used for receiving for streamed messages from particular room. It comes under MessageCallback Context.
I will add example here.

Reply:
I think it needs a better naming, it is not a subscription listener, it listens for messages...